### PR TITLE
[40] - Refactor hexagonal architecture

### LINKED ISSUE #40: Refactor Hexagonal Architecture
Leave the entire project defined in three folders (Application Domain and Infrastructure)

### PR DESCRIPTION
**Pull request by IvanSolaDelgado**
**Issue**
- Resolves #40 

**Solution**
- I have moved the Http folder to Infrastructure since it contains the Form Requests responsible of handling the technical aspects related to the entry and validation of data from abroad.
- I have moved the Validators folder to Infrastructure since there are certain data validations that come from outside (similar to Http).
- I have moved the Console folder to Infrastructure as the Console folder contains custom commands that have to do with the Laravel Infrastructure.

### CODE REVIEW

File: app/Infrastructure/Http/Requests/OpenWalletRequest.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace App\Http\Requests;
+namespace App\Infrastructure\Http\Requests;
```
Comment (AsierAlba10) [inline app/Infrastructure/Http/Requests/OpenWalletRequest.php:3]: It is our standard to organize validators or form requests as close as possible to where they are being used.

In the controllers folder you can extract a controller and its validator to a folder. For example OpenWallet and put both inside.

File: app/Infrastructure/Controllers/PostOpenWalletController.php [hunk 2]
```diff
@@ -3,7 +3,7 @@
 namespace App\Infrastructure\Controllers;
```
Comment (AsierAlba10) [inline app/Infrastructure/Controllers/PostOpenWalletController.php:3]: Do you think this is the best name we could give to the controller? Is the method used in this controller really necessary in the name?

File: app/Infrastructure/Validators/WalletIdValidator.php [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace App\Validators;
+namespace App\Infrastructure\Validators;
```
Comment (AsierAlba10) [inline app/Infrastructure/Validators/WalletIdValidator.php:3]: Same, move this :)

File: tests/app/Infrastructure/Controller/PostOpenWalletControllerTest.php [hunk 4]
```diff
@@ -13,7 +13,6 @@ class PostOpenWalletControllerTest extends TestCase
 {
```
Comment (AsierAlba10) [inline tests/app/Infrastructure/Controller/PostOpenWalletControllerTest.php:13]: If you change the controller name you must change this

File: bootstrap/app.php [hunk 5]
```diff
@@ -33,7 +33,7 @@
 
 $app->singleton(
     Illuminate\Contracts\Console\Kernel::class,
-    App\Console\Kernel::class
+    \App\Infrastructure\Console\Kernel::class
```
Comment (AsierAlba10) [inline bootstrap/app.php:36]: Creo que si borras la primera \ también funciona. Prueba

File: tests/app/Infrastructure/Http/Requests/OpenWalletRequestTest.php [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <?php
 
-namespace Tests\app\Http\Requests;
+namespace Tests\app\Infrastructure\Http\Requests;
```
Comment (AsierAlba10) [inline tests/app/Infrastructure/Http/Requests/OpenWalletRequestTest.php:3]: In addition to sorting the tests by the layers of the hexagonal architecture, we also sort them by the type of test: Unitary, integration or e2e.

**Review by AsierAlba10 — CHANGES_REQUESTED** [review]

File: tests/app/Infrastructure/Http/Requests/OpenWalletRequestTest.php (hunk 6 above)
Comment (IvanSolaDelgado) [reply tests/app/Infrastructure/Http/Requests/OpenWalletRequestTest.php:3]: By this do you mean that we should create a folder for unit tests and another for integration tests?

File: app/Infrastructure/Controllers/BuyCoin/BuyCoinController.php [hunk 7]
```diff
@@ -1,16 +1,16 @@
 <?php
 
-namespace App\Infrastructure\Controllers;
+namespace App\Infrastructure\Controllers\BuyCoin;
 
 use App\Application\Exceptions\CoinNotFoundException;
 use App\Application\Exceptions\WalletNotFoundException;
 use App\Application\Services\BuyCoinService;
-use App\Http\Requests\BuyCoinRequest;
+use App\Infrastructure\Controllers\BuyCoin\BuyCoinRequest;
```
Comment (AsierAlba10) [inline app/Infrastructure/Controllers/BuyCoin/BuyCoinController.php]: Delete

File: app/Infrastructure/Controllers/Validators/WalletIdValidator.php [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace App\Validators;
+namespace App\Infrastructure\Controllers\Validators;
```
Comment (AsierAlba10) [inline app/Infrastructure/Controllers/Validators/WalletIdValidator.php]: If you are using it as a "CustomRequest" why don't you define it as two different classes and add it as close as possible to where you are using it.

File: routes/api.php [hunk 9]
```diff
@@ -1,10 +1,10 @@
...
-use App\Infrastructure\Controllers\PostBuyCoinController;
-use App\Infrastructure\Controllers\PostSellCoinController;
+use App\Infrastructure\Controllers\BuyCoin\BuyCoinController;
+use App\Infrastructure\Controllers\OpenWallet\OpenWalletController;
+use App\Infrastructure\Controllers\SellCoin\SellCoinController;
+use App\Infrastructure\Controllers\WalletBalance\WalletBalanceController;
+use App\Infrastructure\Controllers\WalletCryptocurrencies\WalletCryptocurrenciesController;
 use Illuminate\Support\Facades\Route;
 
 /*
```
Comment (AsierAlba10) [inline routes/api.php:10]: This can be deleted

**Review by AsierAlba10 — APPROVED** [review]

**Pull request merged**